planner/core/planbuilder.go

```diff
@@ -3184,6 +3184,67 @@ func (b *PlanBuilder) getAffectCols(insertStmt *ast.InsertStmt, insertPlan *Inse
 	return affectedValuesCols, nil
 }
 
+func (b PlanBuilder) getInsertColExpr(ctx context.Context, insertPlan *Insert, mockTablePlan *LogicalTableDual, col *table.Column, expr ast.ExprNode, checkRefColumn func(n ast.Node) ast.Node) (outExpr expression.Expression, err error) {
+	if col.Hidden {
+		return nil, ErrUnknownColumn.GenWithStackByArgs(col.Name, clauseMsg[fieldList])
+	}
+	switch x := expr.(type) {
+	case *ast.DefaultExpr:
+		refCol := col
+		if x.Name != nil {
+			found := false
+			for _, col := range insertPlan.Table.Cols() {
+				if col.Name.L == x.Name.Name.L {
+					refCol = col
+					found = true
+					break
+				}
+			}
```

Comment:
We can use `FindColLowerCase`. BTW, should we use `insertPlan.Table.VisibleCols()`?

Reply:
I think it should be all columns, since hidden/generated columns should be OK to use if they refer to their own DEFAULT, like `hidden_generated_col = DEFAULT(hidden_generated_col)`